### PR TITLE
OSSM-849: Update the luajit2 dependency to 1085a4d

### PR DESCRIPTION
**Pull request by knm3000**
- Fixes issues on s390x (https://github.com/openresty/luajit2/commit/1085a4d562b449e7be9e4508b52a19651bdf04a6)

### CODE REVIEW

File: bazel/repository_locations.bzl
Comment (jwendell) [inline bazel/repository_locations.bzl:261]: Do we need to update the release-date field as well?

**Review by jwendell — COMMENTED** [review]
Which issue does this fix? @oschaaf do we have an OSSM issue to track this?

File: bazel/repository_locations.bzl
Comment (knm3000) [reply bazel/repository_locations.bzl:261]: Done

**Comment by oschaaf** [comment]
> Which issue does this fix? @oschaaf do we have an OSSM issue to track this?

@jwendell created https://issues.redhat.com/browse/OSSM-849 for this.

**Review by oschaaf — CHANGES_REQUESTED** [review]
@knm3000 would you be ok with updating the title of this PR to something like:
 `Update the luajit2 dependency to 1085a4d` ..  And then perhaps set the PR/commit description to something like:

```
- Fixes issues on s390x
- ... any other changes worth calling out ..
- ..
```

**Comment by knm3000** [comment]
Updated PR/commit name and description.

**Review by oschaaf — APPROVED** [review]
Thank you @knm3000 for following up. 
LGTM

**Comment by yselkowitz** [comment]
Do we want to do this in 2.1.z or wait until 2.2?  Or IOW, is a luajit rebase planned for 2.1.z?

**Review by oschaaf — CHANGES_REQUESTED** [review]
Sorry, this fell through the cracks, but asking around we do require DCO signoff in this repo.. 
So one more ask to update the commit accordingly. Sorry!
If you need some hints on things around DCO and amending this, please see 
https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#dco-sign-your-work

**Comment by knm3000** [comment]
Added DCO sign-off into the commit

**Review by oschaaf — COMMENTED** [review]
Thanks for adding the DCO

**Comment by yselkowitz** [comment]
For 2.1.0 this is building on s390x with patched luajit.  The switch to luajit2 is only needed upon a rebase of luajit (on which the s390x patch does not apply).  So unless you plan on rebasing luajit in 2.1.z, or there is a bug in the http.lua filter with 2.1 on s390x, waiting to switch until 2.2 seems to be prudent.
/hold

**Comment by twghu** [comment]
The 2.2 wip branch has been updated to include https://github.com/openresty/luajit2/commit/1085a4d562b449e7be9e4508b52a19651bdf04a6

**Comment by knm3000** [comment]
Opened another PR https://github.com/maistra/envoy/pull/158 against maistra-2.2 instead of this one.

**Pull request closed without merging**
